ct_core/include/ct/core/systems/discrete_time/linear/DiscreteLinearSystem.h

```diff
@@ -49,6 +49,44 @@ class DiscreteLinearSystem : public DiscreteControlledSystem<STATE_DIM, CONTROL_
     //! deep cloning
     virtual DiscreteLinearSystem<STATE_DIM, CONTROL_DIM, SCALAR>* clone() const override = 0;
 
+    //! computes the controllability gramian by discretization
+    /*!
+     * Computes the controllability gramian
+     *
+     * @return Returns the controllability Gramian
+     */
+    Eigen::Matrix<double, STATE_DIM, STATE_DIM> computeControllabilityGramian(const size_t max_iters=100,
+        const double tolerance=1e-9)
+    {
+        state_matrix_t A;
+        state_control_matrix_t B;
+
+        // TODO: Need to get the matrices A and B
+        //       Not neccessary to specifiy x and u?
+        // this->getAandB(x, u, n, A, B);
```

Comment:
Wasn't quite sure of this. We probably need to overload `getAandB` so that we can retrieve `A` and `B` matrices for linear systems without explicitly specifying `x`, `u` and `n`?

Reply:
oh I see. Looking at the structure, I realize that actually, there is a class missing on the discrete-time side of things. Missing is the class "DiscreteLTISystem"
So that means we have two options here:
* you either allow the state and control to be handed to the gramian computation method here
* you implement the class DiscreteLTISystem analoguously to the LTISystem.h and put the method there

Reply:
I shifted the methods to the new class `DiscreteLTISystem`. Let me know if any changes required.